[BUG] 0.3版 windows build fail

**發生了什麼事？ / The Problem**
- 在 windows 上 0.3 版跑 go build 會 fail
**如何重現 / To Reproduce**
- go build
**預期的行為 / Expected behavior**
build success

**螢幕截圖 / Screenshots**
```
PS C:\Users\Tim.chen\Documents\GolangProjects\go-bbs\pttbbs> go build
# github.com/Ptt-official-app/go-bbs/cache
..\cache\mmap.go:54:12: undefined: openMmap
..\cache\mmap.go:75:9: undefined: closeMmap
..\cache\systemvshm.go:27:38: undefined: ShmidDs
..\cache\systemvshm.go:58:9: undefined: ShmidDs
..\cache\systemvshm_unix.go:9:16: undefined: unix.Syscall
..\cache\systemvshm_unix.go:9:32: undefined: unix.SYS_SHMGET
..\cache\systemvshm_unix.go:17:16: undefined: unix.Syscall
..\cache\systemvshm_unix.go:17:32: undefined: unix.SYS_SHMAT
..\cache\systemvshm_unix.go:27:16: undefined: unix.Syscall
..\cache\systemvshm_unix.go:36:38: undefined: ShmidDs
..\cache\systemvshm_unix.go:27:16: too many errors
```

**開發環境 / Environment**
 - OS: windows 10
 - Golang Version: 1.16
 - Package Version: 0.3

追加 0.31 版錯誤訊息 

```
PS C:\Users\Tim.chen\Documents\GolangProjects\go-bbs\cache> go build 
# github.com/Ptt-official-app/go-bbs/cache
.\mmap.go:54:12: undefined: openMmap
.\mmap.go:75:9: undefined: closeMmap
.\systemvshm.go:64:16: ds.ShmSegsz undefined (type ShmidDs has no field or method ShmSegsz)
.\systemvshm_unix.go:10:16: undefined: unix.Syscall
.\systemvshm_unix.go:10:32: undefined: unix.SYS_SHMGET
.\systemvshm_unix.go:18:16: undefined: unix.Syscall
.\systemvshm_windows.go:10:6: shmget redeclared in this block
        previous declaration at .\systemvshm_unix.go:9:48
.\systemvshm_windows.go:14:6: shmat redeclared in this block
        previous declaration at .\systemvshm_unix.go:17:62
.\systemvshm_windows.go:18:6: shmdt redeclared in this block
        previous declaration at .\systemvshm_unix.go:27:35
.\systemvshm_windows.go:22:6: shmctl redeclared in this block
        previous declaration at .\systemvshm_unix.go:37:53
.\systemvshm_unix.go:18:16: too many errors
```

--- a/cache/mmap_windows.go
+++ b/cache/mmap_windows.go
@@ -1,0 +1,16 @@
+// +build windows
+
+package cache
+
+import (
+	"fmt"
+)
+
+func openMmap(fd int, size int) ([]byte, error) {
+	return nil, fmt.Errorf("windows do not implement shmget")
+
+}
+
+func closeMmap(buf []byte) error {
+	return fmt.Errorf("windows do not implement shmget")
+}

--- a/cache/mmap_windows.go
+++ b/cache/mmap_windows.go
@@ -7,10 +7,10 @@ import (
 )
 
 func openMmap(fd int, size int) ([]byte, error) {
-	return nil, fmt.Errorf("windows do not implement shmget")
+	return nil, fmt.Errorf("TODO: implement open mmap")
 
 }
 
 func closeMmap(buf []byte) error {
-	return fmt.Errorf("windows do not implement shmget")
+	return fmt.Errorf("TODO: implement close mmap")
 }

--- a/cache/systemvshm_windows.go
+++ b/cache/systemvshm_windows.go
@@ -5,7 +5,11 @@ import (
 	"fmt"
 )
 
-type ShmidDs struct{}
+type sizeT uint64
+
+type ShmidDs struct{
+	ShmSegsz    sizeT
+}
 
 func shmget(key int, size int, flag int) (int, error) {
 	return 0, fmt.Errorf("windows do not implement shmget")
@@ -21,4 +25,8 @@ func shmdt(shmaddr uintptr) (int, error) {
 
 func shmctl(shmid int, cmd int, buf *ShmidDs) (int, error) {
 	return 0, fmt.Errorf("windows do not implement shmctl")
+}
+
+func ShmSegsz(shmid int, cmd int, buf *ShmidDs) (int, error) {
+	return 0, fmt.Errorf("windows do not implement ShmSegsz")
 }

--- a/cache/systemvshm_windows.go
+++ b/cache/systemvshm_windows.go
@@ -1,4 +1,5 @@
 // +build windows
+
 package cache
 
 import (

--- a/cache/systemvshm_windows.go
+++ b/cache/systemvshm_windows.go
@@ -26,7 +26,3 @@ func shmdt(shmaddr uintptr) (int, error) {
 func shmctl(shmid int, cmd int, buf *ShmidDs) (int, error) {
 	return 0, fmt.Errorf("windows do not implement shmctl")
 }
-
-func ShmSegsz(shmid int, cmd int, buf *ShmidDs) (int, error) {
-	return 0, fmt.Errorf("windows do not implement ShmSegsz")
-}

